x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/MountSnapshotStep.java

```diff
@@ -37,17 +39,34 @@ public class MountSnapshotStep extends AsyncRetryDuringSnapshotActionStep {
 
     private final String restoredIndexPrefix;
     private final MountSearchableSnapshotRequest.Storage storageType;
+    @Nullable
+    private final Integer totalShardsPerNode;
 
     public MountSnapshotStep(
         StepKey key,
         StepKey nextStepKey,
         Client client,
         String restoredIndexPrefix,
-        MountSearchableSnapshotRequest.Storage storageType
+        MountSearchableSnapshotRequest.Storage storageType,
+        @Nullable Integer totalShardsPerNode
     ) {
         super(key, nextStepKey, client);
         this.restoredIndexPrefix = restoredIndexPrefix;
         this.storageType = Objects.requireNonNull(storageType, "a storage type must be specified");
+        if (totalShardsPerNode != null && totalShardsPerNode < 1) {
+            throw new IllegalArgumentException("[totalShardsPerNode] must be >= 1");
```

Comment:
I think this should use the preferred name (and to match the config flag name), something like:
```suggestion
            throw new IllegalArgumentException("[" + SearchableSnapshotAction.TOTAL_SHARDS_PER_NODE + "] must be >= 1");
```